### ISSUE TITLE
DJ012 Doesn't Support Cross-Module Inheritance

### ISSUE THREAD

**Issue by max-muoto** [astral-sh/ruff#9291]
In a Django app, you might have some custom base model that all of your other models inherit from.

```python
class BaseModel(models.Model):
    default_field_1 = models.CharField(max_length=255, default="")
    default_field_2 = models.CharField(max_length=255, default="")

    class Meta:
        abstract = True

class ExampleModel(BaseModel):
    def __str__(self) -> str: # Ruff flags as this being incorrectly ordered.
        return super().__str__()

    class Meta:
        ...
```

However, if you split these across modules, Ruff will not flag incorrect ordering in `ExampleModel` with [DJ012](https://docs.astral.sh/ruff/rules/django-unordered-body-content-in-model/) enabled.

Command: `ruff check .`
Version `0.1.9`

**Comment by charliermarsh** [comment]
Thanks! This is part of a much broader project to enable multi-file analysis, so we're just merging generic symptoms of that issue into https://github.com/astral-sh/ruff/issues/7447.

**Comment by max-muoto** [comment]
Sounds good, thanks!

**Comment by mounirmesselmeni** [comment]
@charliermarsh I would suggest to update the rule docs about this limitation.

Only works if you don't use custom base model defined in another file or module.
